### PR TITLE
Fluent-2 V8 Theme: Ensure fluent v2 buttons focus does not bleed over bounding box

### PR DESCRIPTION
**Pull request by Mitch-At-Work**
## Previous Behavior
The focus outline would bleed over the bounding box by 1px

## New Behavior
Focus is now contained within the bounding box/component outline by shifting it inward by 1px.

## Related Issue(s)
N/A (internal)

- Fixes #

### CODE REVIEW

**Comment by Mitch-At-Work** [comment]
It seems that this is an intentional style in fluent V9 - however we do have some issues with the external border causing some edge-bleed as it is not within the bounding box, creating this PR to demonstrate internal width and whether that is desirable.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/595d25e342e8c26ae484fe1c0a477c80e72232a9'>595d25e342e8c26ae484fe1c0a477c80e72232a9</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by Mitch-At-Work** [comment]
Consulted with FluentUI design, we are going to migrate V9 border-thickening to the internal side of a bounding box (in buttons), this will ensure that we do not cause any overlap external to bounds, nor have any movement from external padding.

**Comment by Mitch-At-Work** [comment]
Here is some examples of where the bleed occurred, design of border thickness has not changed, but it is now moved to be internal:
image
image

**Comment by Mitch-At-Work** [comment]
Focus style after changes: 
image

File: change/@fluentui-fluent2-theme-509c93b1-ce92-4baf-bb25-32e296c59e20.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "[BUG] Fix button focus bleeding into outside bounding box",
```
Comment (spmonahan) [inline change/@fluentui-fluent2-theme-509c93b1-ce92-4baf-bb25-32e296c59e20.json:3]:
```suggestion
  "comment": "fix: button focus bleeding into outside bounding box",
```

**Review by spmonahan — APPROVED** [review]

**Pull request merged**